source/common/http/filter/ip_tagging_filter.h

```diff
@@ -5,52 +5,94 @@
 #include <string>
 #include <vector>
 
+#include "envoy/common/exception.h"
+#include "envoy/config/filter/http/ip_tagging/v2/ip_tagging.pb.h"
 #include "envoy/http/filter.h"
-#include "envoy/json/json_object.h"
+#include "envoy/runtime/runtime.h"
 
-#include "common/common/assert.h"
-#include "common/json/config_schemas.h"
-#include "common/json/json_validator.h"
+#include "common/network/cidr_range.h"
+#include "common/network/lc_trie.h"
 
 namespace Envoy {
 namespace Http {
 
 /**
  * Type of requests the filter should apply to.
  */
-enum class FilterRequestType { Internal, External, Both };
+enum class FilterRequestType { INTERNAL, EXTERNAL, BOTH };
 
 /**
- * Configuration for the ip tagging filter.
+ * Configuration for the HTTP IP Tagging filter.
  */
-class IpTaggingFilterConfig : Json::Validator {
+class IpTaggingFilterConfig {
 public:
-  IpTaggingFilterConfig(const Json::Object& json_config)
-      : Json::Validator(json_config, Json::Schema::IP_TAGGING_HTTP_FILTER_SCHEMA),
-        request_type_(stringToType(json_config.getString("request_type", "both"))) {}
+  IpTaggingFilterConfig(const envoy::config::filter::http::ip_tagging::v2::IPTagging& config,
+                        const std::string& stat_prefix, Stats::Scope& scope,
+                        Runtime::Loader& runtime)
+      : request_type_(requestTypeEnum(config.request_type())), scope_(scope), runtime_(runtime),
+        stats_prefix_(stat_prefix + "ip_tagging.") {
+    if (config.ip_tags_size() == 0) {
+      throw EnvoyException("HTTP IP Tagging Filter requires ip_tags to be specified.");
+    }
+
+    std::vector<std::pair<std::string, std::vector<Network::Address::CidrRange>>> tag_data;
+    for (const auto& ip_tag : config.ip_tags()) {
+      std::pair<std::string, std::vector<Network::Address::CidrRange>> ip_tag_pair;
+      ip_tag_pair.first = ip_tag.ip_tag_name();
+
+      std::vector<Network::Address::CidrRange> cidr_set;
+      for (const envoy::api::v2::core::CidrRange& entry : ip_tag.ip_list()) {
+
+        // TODO(ccaraman): Update CidrRange::create to support only constructing valid ranges.
+        // Currently, CidrRange::create doesn't guarantee that the CidrRanges are valid.
+        Network::Address::CidrRange cidr_entry = Network::Address::CidrRange::create(entry);
+        if (cidr_entry.isValid()) {
+          cidr_set.push_back(cidr_entry);
+        } else {
+          throw EnvoyException(
+              fmt::format("invalid ip/mask combo '{}/{}' (format is <ip>/<# mask bits>)",
+                          entry.address_prefix(), entry.prefix_len().value()));
+        }
+      }
+      ip_tag_pair.second = cidr_set;
+      tag_data.push_back(ip_tag_pair);
+    }
+    trie_.reset(new Network::LcTrie::LcTrie(tag_data));
+  }
 
+  Runtime::Loader& runtime() { return runtime_; }
+  Stats::Scope& scope() { return scope_; }
   FilterRequestType requestType() const { return request_type_; }
+  Network::LcTrie::LcTrie& trie() const { return *trie_; }
```

Comment:
The return type should be `const Network::LcTrie::LcTrie&` since  callers shouldn't be modifying the trie.